### PR TITLE
[benchmark_app] remove the device restriction when loading extension library

### PR DESCRIPTION
**Pull request by wdkwyf**
### Details:
As for custom GPU operation, user need the ngraph register library(can be loaded via `-l xxx.dll`) and the GPU config xml (can be loaded via `-c xxx.xml`).
But, currently if user choose `-d GPU`, extension library will not be loaded, which casue the `readnetwork() `fail.
Therefore, user have to use the` -d HETERO:GPU,CPU` to enable extension library loading, but this heteroplugin indeed has different behavior with `-d GPU.`

### CODE REVIEW

File: samples/cpp/benchmark_app/main.cpp [hunk 1]
```diff
@@ -189,7 +189,7 @@ int main(int argc, char* argv[]) {
         next_step();
 
         Core ie;
-        if (FLAGS_d.find("CPU") != std::string::npos && !FLAGS_l.empty()) {
+        if (FLAGS_l.empty()) {
```
Comment (nkogteva) [inline samples/cpp/benchmark_app/main.cpp:192]: I don't understand this change. Maybe it should be
```suggestion
        if (!FLAGS_l.empty()) {
```

Comment (wdkwyf) [reply samples/cpp/benchmark_app/main.cpp:192]: Hi, sorry it's my mistake.

**Comment by vladimir-dudnik** [comment]
cc @fzhar @nignatovsky

**Comment by wdkwyf** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
@wdkwyf please rebase (recently PR with dynamic models support was merged)

**Pull request closed without merging**